### PR TITLE
Move over most remaining uses of indexSettings(Version) to IndexVersion

### PR DESCRIPTION
**Pull request by thecoop**
Move most uses of `ESTestCase.indexSettings(Version)` to use `IndexVersion` instead. Note the new method in `IndexVersionUtils`, which assumes a (light) semver policy that wasn't previously specified - but is likely to be followed in the future.

There are a couple more complicated uses still remaining

### CODE REVIEW

File: modules/reindex/src/internalClusterTest/java/org/elasticsearch/migration/AbstractFeatureMigrationIntegTest.java [hunk 1]
```diff
@@ -66,6 +68,9 @@ public abstract class AbstractFeatureMigrationIntegTest extends ESIntegTestCase
     static final int INDEX_DOC_COUNT = 100; // arbitrarily chosen
     static final int INTERNAL_MANAGED_FLAG_VALUE = 1;
     public static final Version NEEDS_UPGRADE_VERSION = TransportGetFeatureUpgradeStatusAction.NO_UPGRADE_REQUIRED_VERSION.previousMajor();
```
Comment (ldematte) [inline modules/reindex/src/internalClusterTest/java/org/elasticsearch/migration/AbstractFeatureMigrationIntegTest.java:70]: Do we still need this Version field, or are we able to move all `NEEDS_UPGRADE_VERSION` usages to `NEEDS_UPGRADE_INDEX_VERSION` and we can remove it?

File: server/src/test/java/org/elasticsearch/cluster/coordination/NodeJoinExecutorTests.java [hunk 2]
```diff
@@ -100,7 +100,7 @@ public void testPreventJoinClusterWithUnsupportedIndices() {
         Settings.builder().build();
         Metadata.Builder metaBuilder = Metadata.builder();
         IndexMetadata indexMetadata = IndexMetadata.builder("test")
-            .settings(settings(Version.fromString("6.8.0"))) // latest V6 released version
+            .settings(settings(IndexVersion.fromId(6080099))) // latest V6 released version
```
Comment (ldematte) [inline server/src/test/java/org/elasticsearch/cluster/coordination/NodeJoinExecutorTests.java:103]: Nit/question: using underscores like in IndexVersion could improve readability/searchabilty? `6_08_00_99`

File: test/framework/src/main/java/org/elasticsearch/test/index/IndexVersionUtils.java [hunk 3]
```diff
@@ -92,6 +92,10 @@ public static IndexVersion getPreviousVersion(IndexVersion version) {
         return ALL_VERSIONS.get(place - 1);
     }
 
+    public static IndexVersion getPreviousMajorVersion(IndexVersion version) {
+        return IndexVersion.fromId(((version.id() / 1_000_000) - 1) * 1_000_000);
```
Comment (ldematte) [inline test/framework/src/main/java/org/elasticsearch/test/index/IndexVersionUtils.java:96]: Shouldn't this also add `99` like in https://github.com/thecoop/elasticsearch/blob/2af22faa54b526c2a0c8688e415f1bfbfa49f374/server/src/main/java/org/elasticsearch/Version.java#L310? So something like `7_14_00_99` becomes `8_00_00_99` -> `V_8_0_0`?

File: modules/reindex/src/internalClusterTest/java/org/elasticsearch/migration/AbstractFeatureMigrationIntegTest.java (hunk 1 above)
Comment (thecoop) [reply modules/reindex/src/internalClusterTest/java/org/elasticsearch/migration/AbstractFeatureMigrationIntegTest.java:70]: The release `Version` is independent of the `IndexVersion`, so we need both here

File: test/framework/src/main/java/org/elasticsearch/test/index/IndexVersionUtils.java (hunk 3 above)
Comment (thecoop) [reply test/framework/src/main/java/org/elasticsearch/test/index/IndexVersionUtils.java:96]: Ah yes, but thats not going to hold for ever - it wont be the case for v9 onwards. I'll work round it

File: server/src/test/java/org/elasticsearch/cluster/coordination/NodeJoinExecutorTests.java (hunk 2 above)
Comment (thecoop) [reply server/src/test/java/org/elasticsearch/cluster/coordination/NodeJoinExecutorTests.java:103]: We havent done that in constants elsewhere - eg `Queries.NESTED_DOCS_IDENTIFIED_VIA_PRIMARY_TERMS_VERSION`. These hardcoded numbers tend to be write-once-read-never anyway...

File: modules/reindex/src/internalClusterTest/java/org/elasticsearch/migration/AbstractFeatureMigrationIntegTest.java (hunk 1 above)
Comment (ldematte) [reply modules/reindex/src/internalClusterTest/java/org/elasticsearch/migration/AbstractFeatureMigrationIntegTest.java:70]: Good to know, thanks

File: test/framework/src/main/java/org/elasticsearch/test/index/IndexVersionUtils.java (hunk 3 above)
Comment (ldematte) [reply test/framework/src/main/java/org/elasticsearch/test/index/IndexVersionUtils.java]: Just for my knowledge/curiosity: do you know why 99 was added, up to now? Any historical reason?

Comment (thecoop) [reply test/framework/src/main/java/org/elasticsearch/test/index/IndexVersionUtils.java]: See the comments in `Version` - it used to represent a final release version, back when there were alpha & beta releases. That hasn't been the case for a long time, so '99' just represents a version that is released.

**Review by ldematte — APPROVED** [review]
LGTM

**Pull request merged**